tools/report-converter/codechecker_report_converter/spotbugs/output_parser.py

```diff
@@ -0,0 +1,171 @@
+#!/usr/bin/env python
+# -------------------------------------------------------------------------
+#                     The CodeChecker Infrastructure
+#   This file is distributed under the University of Illinois Open Source
+#   License. See LICENSE.TXT for details.
+# -------------------------------------------------------------------------
+
+import logging
+import os
+import xml.etree.ElementTree as ET
+
+from ..output_parser import Message, Event, BaseParser
+
+LOG = logging.getLogger('ReportConverter')
+
+
+class SpotBugsMessage(Message):
+    """ Represents a message with an optional event, fixit and note messages.
+
+    This will be a diagnostic section in the plist which represents a report.
+    """
+
+    def __init__(self, path, line, column, message, checker, report_hash,
+                 events=None, notes=None, fixits=None):
+        super(SpotBugsMessage, self).__init__(path, line, column, message,
+                                              checker, events, notes, fixits)
+        self.report_hash = report_hash
+
+    def __eq__(self, other):
+        return super(SpotBugsMessage, self).__eq__(other) and \
+            self.report_hash == other.report_hash
+
+    def __str__(self):
+        return '%s, report_hash=%s' % \
+               (super(SpotBugsMessage, self).__str__(), self.report_hash)
+
+
+class SpotBugsParser(BaseParser):
+    """ Parser for SpotBugs output. """
+
+    def __init__(self):
+        super(SpotBugsParser, self).__init__()
+        self.project_paths = []
+
+    def parse_messages(self, analyzer_result):
+        """ Parse the given analyzer result. """
+
+        root = self.__parse_analyzer_result(analyzer_result)
+        if root is None:
+            return
+
+        self.project_paths = self.__get_project_paths(root)
+
+        for bug in root.findall('BugInstance'):
+            message = self.__parse_bug(bug)
+            if message:
+                self.messages.append(message)
+
+        return self.messages
+
+    def __get_abs_path(self, source_path):
+        """ Returns full path of the given source path.
+
+        It will try to find the given source path in the project paths and
+        returns full path if it founds.
+        """
+        if os.path.exists(source_path):
+            return source_path
+
+        for project_path in self.project_paths:
+            full_path = os.path.join(project_path, source_path)
+            if os.path.exists(full_path):
+                return full_path
+
+        LOG.warning("No source file found: %s", source_path)
+
+    def __parse_analyzer_result(self, analyzer_result):
+        """ Parse the given analyzer result xml file.
+
+        Returns the root element of the parsed tree or None if something goes
+        wrong.
+        """
+        try:
+            tree = ET.parse(analyzer_result)
+            return tree.getroot()
+        except IOError:
```

Comment:
IOError was merged to OSError, does parse throw an IOError exception?
https://docs.python.org/3/library/exceptions.html
```
Changed in version 3.3: EnvironmentError, IOError, WindowsError, socket.error, select.error and mmap.error have been merged into OSError, and the constructor may return a subclass.
```